src/Meziantou.Analyzer/Rules/AwaitAwaitableMethodInSyncMethodAnalyzer.cs

```diff
@@ -30,9 +30,6 @@ public override void Initialize(AnalysisContext context)
             var awaitableTypes = new AwaitableTypes(context.Compilation);
             context.RegisterSymbolStartAction(context =>
             {
-                if (context.Symbol is IMethodSymbol method && (method.IsAsync || method.IsTopLevelStatementsEntryPointMethod()))
-                    return; // Already handled by CS4014
```

Comment:
Can you move the comment `// Already handled by CS4014` to the line 58?

Also, you can remove the `// While there is a check in RegisterSymbolStartAction, this is needed to handle lambda and delegates` comment as it is not valid anymore.

Reply:
Good catch. Comments fixed.